Common/itkComputeImageExtremaFilter.hxx

```diff
@@ -176,7 +176,9 @@ ComputeImageExtremaFilter<TInputImage>::ThreadedGenerateDataImageSpatialMask(con
     } // end for
   }   // end if
 
+#ifndef __wasi__
   std::lock_guard<std::mutex> mutexHolder(m_Mutex);
+#endif
```

Comment:
Does WASI support multi-threading? Specifically, will this ComputeImageExtremaFilter run using multiple threads in parallel?

Reply:
Currently, the default WASI toolchain does not support multi-threading. This is being improved, but we need these for now.

Reply:
Sorry, still not very happy about having `#ifndef __wasi__` all over the place. Would it be an idea to have a "dummy" mutex class (that does not do anything) for single-threaded builds.

    #ifdef ELX_SINGLE_THREADED_BUILD
      using ElastixMutexType = DummyMutex;  // Mutex-like dummy type that does nothing
    #else
      using ElastixMutexType = std::mutex;
    #endif

Reply:
Sorry, I do not see how a dummy mutex type addresses the `std::lock_guard` and the mutex class members.

Reply:
`std::lock_guard` should be able to support any "mutex-like" type, that has a `lock()` and an `unlock()` member function. So for example, would the following code compile on WASI? https://godbolt.org/z/oP5T6Pn35  Can you please try that out?

```
#define ELX_SINGLE_THREADED // For WebAssembly Support

#include <mutex>

// Mutex for single-threading, does nothing.
class SingleThreadedMutex
{
  public:
    void lock() {}
    void unlock() {}
};

// The MutexType type alias:
#ifdef ELX_SINGLE_THREADED
using MutexType = SingleThreadedMutex;
#else
using MutexType = std::mutex;
#endif

// Use case:
MutexType mutexVariable;
std::lock_guard<MutexType> lockGuard(mutexVariable);

```

Reply:
@thewtex Would it be helpful to you if I would implement such a `SingleThreadedMutex` for elastix, that would replace `std::mutex` when a CMake option like `ELX_SINGLE_THREADED` would be ON?